### ISSUE TITLE
[KED-2474] Error installing kedro-airflow with pip

### ISSUE THREAD

**Issue by younggalois** [quantumblacklabs/kedro-airflow#66]
## Description
Hello! I am getting a pip dependency resolution error when installing kedro-airflow in a clean Conda environment.

## Context
I am unable to successfully install the library. I've tried with python 3.6, 3.7, and 3.8 but get the similar error each time

## Steps to Reproduce

1. ```conda create -n satoshi5 python=3.7```
2. ```conda activate satoshi5```
3. ```pip install kedro-airflow```

## Expected Result
installation should success

## Actual Result
receive the following error message:
```
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

marshmallow-sqlalchemy 0.24.1 requires marshmallow>=3.0.0, but you'll have marshmallow 2.21.0 which is incompatible.
cattrs 1.1.1 requires attrs>=20.1.0, but you'll have attrs 19.3.0 which is incompatible.
```

The info about the new pip resolver might be relevant, but my pip version is 20.2 and the new version is the default in 20.3 so I don't think it is actually using it.

## Your Environment
created the environment with conda as in the above code

* Kedro-Airflow plugin version used (get it by running `pip show kedro-airflow`):
```
Name: kedro-airflow
Version: 0.3.0
Summary: Kedro-Airflow makes it easy to deploy Kedro projects to Airflow
Home-page: https://github.com/quantumblacklabs/kedro-airflow
Author: QuantumBlack Labs
Author-email: None
License: Apache Software License (Apache 2.0)
Location: /Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages
Requires: kedro, python-slugify, semver, apache-airflow
Required-by:
```
* Airflow version (`airflow --version`):
resulted in the following error:
```
Traceback (most recent call last):
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/bin/airflow", line 26, in <module>
    from airflow.bin.cli import CLIFactory
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/airflow/bin/cli.py", line 94, in <module>
    api_module = import_module(conf.get('cli', 'api_client'))  # type: Any
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/airflow/api/client/local_client.py", line 24, in <module>
    from airflow.api.common.experimental import delete_dag
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/airflow/api/common/experimental/delete_dag.py", line 26, in <module>
    from airflow.models.serialized_dag import SerializedDagModel
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/airflow/models/serialized_dag.py", line 35, in <module>
    from airflow.serialization.serialized_objects import SerializedDAG
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/airflow/serialization/serialized_objects.py", line 28, in <module>
    import cattr
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/cattr/__init__.py", line 1, in <module>
    from .converters import Converter, GenConverter, UnstructureStrategy
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/cattr/converters.py", line 16, in <module>
    from attr import fields, resolve_types
ImportError: cannot import name 'resolve_types' from 'attr' (/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/attr/__init__.py)
```
* Kedro version used (`pip show kedro` or `kedro -V`):
resulted in the following error:
```
Traceback (most recent call last):
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/kedro/framework/cli/cli.py", line 683, in load_entry_points
    entry_point_commands.append(entry_point.load())
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2446, in load
    self.require(*args, **kwargs)
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2469, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 775, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (attrs 19.3.0 (/Users/tyler/opt/anaconda3/envs/satoshi5/lib/python3.7/site-packages), Requirement.parse('attrs>=20.1.0'), {'cattrs'})
Error: Loading project commands from airflow = kedro_airflow.plugin:commands
kedro, version 0.16.6
```
* Python version used (`python -V`): Python 3.7.9

* Operating system and version: MacOS v10.14.6

Here is my full conda environment as well after installing kedro-airflow:
```
# packages in environment at /Users/tyler/opt/anaconda3/envs/satoshi5:
#
# Name                    Version                   Build  Channel
alembic                   1.4.3                    pypi_0    pypi
anyconfig                 0.9.11                   pypi_0    pypi
apache-airflow            1.10.12                  pypi_0    pypi
apispec                   1.3.3                    pypi_0    pypi
argcomplete               1.12.2                   pypi_0    pypi
arrow                     0.17.0                   pypi_0    pypi
attrs                     19.3.0                   pypi_0    pypi
babel                     2.9.0                    pypi_0    pypi
binaryornot               0.4.4                    pypi_0    pypi
ca-certificates           2020.10.14                    0  
cached-property           1.5.2                    pypi_0    pypi
cachetools                4.1.1                    pypi_0    pypi
cattrs                    1.1.1                    pypi_0    pypi
certifi                   2020.11.8        py37hecd8cb5_0  
chardet                   3.0.4                    pypi_0    pypi
click                     7.1.2                    pypi_0    pypi
colorama                  0.4.4                    pypi_0    pypi
colorlog                  4.0.2                    pypi_0    pypi
configparser              3.5.3                    pypi_0    pypi
cookiecutter              1.6.0                    pypi_0    pypi
croniter                  0.3.36                   pypi_0    pypi
defusedxml                0.6.0                    pypi_0    pypi
dill                      0.3.3                    pypi_0    pypi
dnspython                 2.0.0                    pypi_0    pypi
docutils                  0.16                     pypi_0    pypi
email-validator           1.1.2                    pypi_0    pypi
flask                     1.1.2                    pypi_0    pypi
flask-admin               1.5.4                    pypi_0    pypi
flask-appbuilder          2.3.4                    pypi_0    pypi
flask-babel               1.0.0                    pypi_0    pypi
flask-caching             1.3.3                    pypi_0    pypi
flask-jwt-extended        3.25.0                   pypi_0    pypi
flask-login               0.4.1                    pypi_0    pypi
flask-openid              1.2.5                    pypi_0    pypi
flask-sqlalchemy          2.4.4                    pypi_0    pypi
flask-swagger             0.2.14                   pypi_0    pypi
flask-wtf                 0.14.3                   pypi_0    pypi
fsspec                    0.6.3                    pypi_0    pypi
funcsigs                  1.0.2                    pypi_0    pypi
future                    0.18.2                   pypi_0    pypi
gunicorn                  20.0.4                   pypi_0    pypi
idna                      2.10                     pypi_0    pypi
importlib-metadata        3.1.0                    pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
iso8601                   0.1.13                   pypi_0    pypi
itsdangerous              1.1.0                    pypi_0    pypi
jinja2                    2.11.2                   pypi_0    pypi
jinja2-time               0.2.0                    pypi_0    pypi
jmespath                  0.10.0                   pypi_0    pypi
json-merge-patch          0.2                      pypi_0    pypi
jsonschema                3.2.0                    pypi_0    pypi
jupyter-client            6.1.7                    pypi_0    pypi
jupyter-core              4.7.0                    pypi_0    pypi
kedro                     0.16.6                   pypi_0    pypi
kedro-airflow             0.3.0                    pypi_0    pypi
lazy-object-proxy         1.5.1                    pypi_0    pypi
libcxx                    10.0.0                        1  
libedit                   3.1.20191231         h1de35cc_1  
libffi                    3.3                  hb1e8313_2  
lockfile                  0.12.2                   pypi_0    pypi
mako                      1.1.3                    pypi_0    pypi
markdown                  2.6.11                   pypi_0    pypi
markupsafe                1.1.1                    pypi_0    pypi
marshmallow               2.21.0                   pypi_0    pypi
marshmallow-enum          1.5.1                    pypi_0    pypi
marshmallow-sqlalchemy    0.24.1                   pypi_0    pypi
natsort                   7.1.0                    pypi_0    pypi
ncurses                   6.2                  h0a44026_1  
numpy                     1.19.4                   pypi_0    pypi
openssl                   1.1.1h               haf1e3a3_0  
pandas                    1.1.4                    pypi_0    pypi
pendulum                  1.4.4                    pypi_0    pypi
pip                       20.2.4           py37hecd8cb5_0  
pip-tools                 5.4.0                    pypi_0    pypi
pluggy                    0.13.1                   pypi_0    pypi
poyo                      0.5.0                    pypi_0    pypi
prison                    0.1.3                    pypi_0    pypi
psutil                    5.7.3                    pypi_0    pypi
pygments                  2.7.2                    pypi_0    pypi
pyjwt                     1.7.1                    pypi_0    pypi
pyrsistent                0.17.3                   pypi_0    pypi
python                    3.7.9                h26836e1_0  
python-daemon             2.2.4                    pypi_0    pypi
python-dateutil           2.8.1                    pypi_0    pypi
python-editor             1.0.4                    pypi_0    pypi
python-graphviz           0.15                     pypi_0    pypi
python-json-logger        0.1.11                   pypi_0    pypi
python-nvd3               0.15.0                   pypi_0    pypi
python-slugify            3.0.3                    pypi_0    pypi
python3-openid            3.2.0                    pypi_0    pypi
pytz                      2020.4                   pypi_0    pypi
pytzdata                  2020.1                   pypi_0    pypi
pyyaml                    5.3.1                    pypi_0    pypi
pyzmq                     20.0.0                   pypi_0    pypi
readline                  8.0                  h1de35cc_0  
requests                  2.25.0                   pypi_0    pypi
semver                    2.13.0                   pypi_0    pypi
setproctitle              1.1.10                   pypi_0    pypi
setuptools                50.3.1           py37hecd8cb5_1  
six                       1.15.0                   pypi_0    pypi
sqlalchemy                1.3.20                   pypi_0    pypi
sqlalchemy-jsonfield      0.9.0                    pypi_0    pypi
sqlalchemy-utils          0.36.8                   pypi_0    pypi
sqlite                    3.33.0               hffcf06c_0  
tabulate                  0.8.7                    pypi_0    pypi
tenacity                  4.12.0                   pypi_0    pypi
text-unidecode            1.2                      pypi_0    pypi
thrift                    0.13.0                   pypi_0    pypi
tk                        8.6.10               hb0a8c7a_0  
toposort                  1.5                      pypi_0    pypi
tornado                   6.1                      pypi_0    pypi
traitlets                 5.0.5                    pypi_0    pypi
typing-extensions         3.7.4.3                  pypi_0    pypi
tzlocal                   1.5.1                    pypi_0    pypi
unicodecsv                0.14.1                   pypi_0    pypi
urllib3                   1.26.2                   pypi_0    pypi
werkzeug                  0.16.1                   pypi_0    pypi
wheel                     0.35.1             pyhd3eb1b0_0  
whichcraft                0.6.1                    pypi_0    pypi
wtforms                   2.3.3                    pypi_0    pypi
xz                        5.2.5                h1de35cc_0  
zipp                      3.4.0                    pypi_0    pypi
zlib                      1.2.11               h1de35cc_3  
zope-deprecation          4.4.0                    pypi_0    pypi
```

**Comment by samirsaliba** [comment]
Also getting similar errors with a fresh venv environment.

Steps:
1. python3 -m venv kedro_aiflow
2. source kedro_aiflow/bin/activate
3. pip install kedro-airflow

Error message:
```
ERROR: flask-jwt-extended 3.25.1 has requirement PyJWT<2.0,>=1.6.4, but you'll have pyjwt 2.0.1 which is incompatible.
ERROR: flask-appbuilder 2.3.4 has requirement marshmallow<3.0.0,>=2.18.0, but you'll have marshmallow 3.10.0 which is incompatible.
```
Running kedro --version works fine, but airflow --version throws an error:

> airflow --version
```
Traceback (most recent call last):
  File "/home/samir/venvs/kedro_aiflow/bin/airflow", line 25, in <module>
    from airflow.configuration import conf
  File "/home/samir/venvs/kedro_aiflow/lib/python3.8/site-packages/airflow/__init__.py", line 50, in <module>
    from airflow.models import DAG  # noqa: E402
  File "/home/samir/venvs/kedro_aiflow/lib/python3.8/site-packages/airflow/models/__init__.py", line 21, in <module>
    from airflow.models.baseoperator import BaseOperator, BaseOperatorLink  # noqa: F401
  File "/home/samir/venvs/kedro_aiflow/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 43, in <module>
    from airflow.models.dag import DAG
  File "/home/samir/venvs/kedro_aiflow/lib/python3.8/site-packages/airflow/models/dag.py", line 52, in <module>
    from airflow.models.dagbag import DagBag
  File "/home/samir/venvs/kedro_aiflow/lib/python3.8/site-packages/airflow/models/dagbag.py", line 50, in <module>
    class DagBag(BaseDagBag, LoggingMixin):
  File "/home/samir/venvs/kedro_aiflow/lib/python3.8/site-packages/airflow/models/dagbag.py", line 80, in DagBag
    DAGBAG_IMPORT_TIMEOUT = conf.getint('core', 'DAGBAG_IMPORT_TIMEOUT')
  File "/home/samir/venvs/kedro_aiflow/lib/python3.8/site-packages/airflow/configuration.py", line 418, in getint
    return int(self.get(section, key, **kwargs))
ValueError: invalid literal for int() with base 10: '30.0'

```

**Comment by idanov** [comment]
@younggalois The 0.3.0 version of the plugin used to work with an older version with Airflow, but newer versions have a clash for some of the dependencies. Could you please check if 0.4.0 works for you?

@samirsaliba What version of `kedro`, `kedro-airflow` and `airflow` do you have installed in your environment? 0.4.0 should work just fine and in fact it shouldn't even require `airflow` as a package in order to generate the dag file (we have forgotten to remove it from the requirement). Could you try running `kedro airflow` from an environment without `airflow` and then `airflow` from an environment without `kedro-airflow` in it? Does that work for you?

**Comment by lorenabalan** [comment]
Closing this as resolved in #74
